src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -1446,23 +1465,14 @@ private bool ParseCommand(string[] args, ref int i, bool noexitSeen, bool isEnco
             }
             else
             {
-                // Collect the remaining parameters and combine them into a single command to be run.
-
-                StringBuilder cmdLineCmdSB = new StringBuilder();
-
-                while (i < args.Length)
+                if (_commandHasArgs)
                 {
-                    cmdLineCmdSB.Append(args[i] + " ");
-                    ++i;
+                    _commandLineCommand = args[i];
                 }
-
-                if (cmdLineCmdSB.Length > 0)
+                else
                 {
-                    // remove the last blank
-                    cmdLineCmdSB.Remove(cmdLineCmdSB.Length - 1, 1);
+                    _commandLineCommand = string.Join(' ', args, i, args.Length - i);                
```

Comment:
`i` is by ref parameter. I guess we still need to add `i = args.Length;`.

Reply:
I see